plugins/modules/system/filesystem.py

```diff
@@ -155,39 +169,39 @@ def fstype(self):
         return type(self).__name__
 
     def get_fs_size(self, dev):
-        """ Return size in bytes of filesystem on device. Returns int """
+        """Return size in bytes of filesystem on device (integer).
+           Should query the info with a per-fstype command that can access the
+           device whenever it is mounted or not, and parse the command output.
+           Parser must ensure to return an integer, or raise a ValueError.
+        """
         raise NotImplementedError()
 
     def create(self, opts, dev):
         if self.module.check_mode:
             return
 
         mkfs = self.module.get_bin_path(self.MKFS, required=True)
-        if opts is None:
-            cmd = "%s %s '%s'" % (mkfs, self.MKFS_FORCE_FLAGS, dev)
-        else:
-            cmd = "%s %s %s '%s'" % (mkfs, self.MKFS_FORCE_FLAGS, opts, dev)
+        cmd = [mkfs] + self.MKFS_FORCE_FLAGS + opts + [str(dev)]
         self.module.run_command(cmd, check_rc=True)
 
     def wipefs(self, dev):
-        if platform.system() == 'FreeBSD':
-            msg = "module param state=absent is currently not supported on this OS (FreeBSD)."
-            self.module.fail_json(msg=msg)
-
         if self.module.check_mode:
             return
 
         # wipefs comes with util-linux package (as 'blockdev' & 'findmnt' above)
-        # so it is not supported on FreeBSD. Even the use of dd as a fallback is
+        # that is ported to FreeBSD. The use of dd as a portable fallback is
         # not doable here if it needs get_mountpoint() (to prevent corruption of
-        # a mounted filesystem), since 'findmnt' is not available on FreeBSD.
+        # a mounted filesystem), since 'findmnt' is not available on FreeBSD,
+        # even in util-linux port for this OS.
         wipefs = self.module.get_bin_path('wipefs', required=True)
```

Comment:
This is for a potential follow-up PR: I think the `get_bin_path()` calls should come *before* the `if self.module.check_mode` check, so that check mode will already indicate that the module cannot find the tools it needs.